### PR TITLE
Add Sigkill handling to node binary

### PR DESCRIPTION
**Pull request by danielSanchezQ**

### CODE REVIEW

File: nomos-services/system-sig/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+pub struct SystemSig {
+    service_state: ServiceStateHandle<Self>,
+}
+
+impl SystemSig {
+    async fn should_stop_service(msg: LifecycleMessage) -> bool {
+        match msg {
+            LifecycleMessage::Shutdown(sender) => {
+                // TODO: Try to finish pending transactions if any and close connections properly
```
Comment (bacv) [inline nomos-services/system-sig/src/lib.rs:23]: I believe there won't be any pending txs in this service

**Review by bacv — APPROVED** [review]
Looks good, just double checking it that todo comment is a leftover.

File: nomos-services/system-sig/src/lib.rs (hunk 1 above)
Comment (danielSanchezQ) [reply nomos-services/system-sig/src/lib.rs:23]: hopefully not 😅

**Review by youngjoon-lee — APPROVED** [review]

**Comment by danielSanchezQ** [comment]
CI codeconv was unrelated

**Pull request merged**